versioned_docs/version-4.6/developers/applications/data-loader.md

```diff
@@ -4,178 +4,122 @@ title: Data Loader
+You can confirm the data is there by hitting the endpoint you created earlier:
 
-1. The Data Loader reads all specified data files (JSON or YAML)
-1. For each file, it validates that a single table is specified
-1. Records are inserted or updated based on timestamp comparison:
-   - New records are inserted if they don't exist
-   - Existing records are updated only if the data file's modification time is newer than the record's updated time
-   - This ensures data files can be safely reloaded without overwriting newer changes
-1. If records with the same primary key already exist, updates occur only when the file is newer
-
-Note: While the Data Loader can create tables automatically by inferring the schema from the provided records, it's recommended to define your table schemas explicitly using the [graphqlSchema](../applications/defining-schemas) component for better control and type safety.
-
-## Best Practices
+```bash
+curl http://localhost:9926/Dog/
+```
 
-1. **Define Schemas First**: While the Data Loader can infer schemas, it's strongly recommended to define your table schemas and relations explicitly using the [graphqlSchema](../applications/defining-schemas) component before loading data. This ensures proper data types, constraints, and relationships between tables.
+You should see both `Harper` and `Balto` returned as JSON.
 
-1. **One Table Per File**: Remember that each data file can only load records into a single table. Organize your files accordingly.
+### Updating Records
 
-1. **Idempotency**: Design your data files to be idempotent - they should be safe to load multiple times without creating duplicate or conflicting data.
+What happens if you change the data file? Let’s update Harper’s age from 3 to 4 in `dogs.json.`
 
-1. **Version Control**: Include your data files in version control to ensure consistency across deployments.
+```json
+{
+	"id": 1,
+	"name": "Harper",
+	"breed": "Labrador",
+	"age": 4,
+	"tricks": ["sit"]
+}
+```
 
-1. **Environment-Specific Data**: Consider using different data files for different environments (development, staging, production).
+When you save the file, Harper will notice the change and reload. The next time you query the endpoint, Harper’s age will be updated.
 
-1. **Data Validation**: Ensure your data files are valid JSON or YAML and match your table schemas before deployment.
+The Data Loader is designed to be safe and repeatable. If a record already exists, it will only update when the file is newer than the record. This means you can re-run deployments without worrying about duplicates.
 
-1. **Sensitive Data**: Avoid including sensitive data like passwords or API keys directly in data files. Use environment variables or secure configuration management instead.
+### Adding More Tables
 
-## Example Component Structure
+If your app grows and you want to seed more than just dogs, you can create additional files. For example, a `settings.yaml` file:
```

Comment:
Did we define a settings table before? I think there is a Breeds table already apart of the guide you could build off of instead.